dsproxy: allow multiple access tokens per datasource

daniellee opened the pull request:
If a plugin has multiple routes for token auth, the token cache will only save one of the tokens. This PR changes the cache key for tokens to cache on datasource id + route path + http method instead of just datasource id.

Fixes #12236

daniellee committed:
dsproxy: allow multiple access tokens per datasource

Changes the cache key for tokens to cache on datasource id +
route path + http method instead of just datasource id.

marefr commented on pkg/api/pluginproxy/ds_proxy.go:
```diff
@@ -57,6 +58,10 @@ func NewDataSourceProxy(ds *m.DataSource, plugin *plugins.DataSourcePlugin, ctx
 		ctx:       ctx,
 		proxyPath: proxyPath,
 		targetUrl: targetURL,
+		httpClient: &http.Client{
```
One http client per proxy vs one global http client earlier - good or bad?

marefr commented on pkg/api/pluginproxy/ds_proxy.go:
```diff
@@ -350,12 +355,16 @@ func (proxy *DataSourceProxy) getAccessToken(data templateData) (string, error)
 }
 
+func (proxy *DataSourceProxy) getAccessTokenCacheKey() string {
+	return fmt.Sprintf("%v_%v_%v", proxy.ds.Id, proxy.route.Path, proxy.route.Method)
```
Possible nil pointer exception if proxy doesn't have a route? Shouldn't be able for this code to be called if there are no route with tokenauth enabled I think so probably not an issue.

marefr reviewed (changes requested):
See my comments

daniellee committed:
dsproxy: move http client variable back

After refactoring to be able to mock the client in
a test, the client wasn't a global variable anymore.
This change moves the client back to being a package-
level variable.
